server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/PrepBulkImport.java

```diff
@@ -209,7 +210,8 @@ private KeyExtent checkForMerge(final long tid, final Manager manager) throws Ex
         try (TabletsMetadata tabletsMetadata =
             TabletsMetadata.builder(manager.getContext()).forTable(bulkInfo.tableId)
                 .overlapping(startRow, null).checkConsistency().fetch(PREV_ROW).build()) {
-          return tabletsMetadata.stream().map(TabletMetadata::getExtent).iterator();
+          return tabletsMetadata.stream().map(TabletMetadata::getExtent)
```

Comment:
Would be nice to avoid buffering the data to a list. Could do the following.

 * Modify TabletIterFactory to make it closeable
 * Make the TabletIterFactory impl track everything it needs to close in a list
 * In TabletIterFactory.close, close everything in the list.
 * Create the TabletIterFactory in the existnig try w/ resources in the code.

Reply:
Addressed in e4c3dad